File: src/endstone_primebds/commands/Server_Management/reloadscripts.py
from endstone import Player
from endstone.command import CommandSender
from endstone_primebds.utils.commandUtil import create_command


from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from endstone_primebds.primebds import PrimeBDS

# Register command
command, permission = create_command(
    "reloadscripts",
    "Reloads the server scripts!",
    ["/reloadscripts"],
    ["primebds.command.reloadscripts"],
    "op",
    ["rscripts", "rs"]
)

# RELOADSCRIPTS COMMAND FUNCTIONALITY
def handler(self: "PrimeBDS", sender: CommandSender, args: list[str]) -> bool:
    self.server.reload_data()
    sender.send_message(f"Server scripts were reloaded!")
    return True

File: src/commands/server/reloadscripts.cpp
/// @file reloadscripts.cpp
/// Reloads server scripts!

#include "primebds/commands/command_registry.h"
#include "primebds/plugin.h"

namespace primebds::commands {

    static bool cmd_reloadscripts(PrimeBDS &, endstone::CommandSender &,
                        const std::vector<std::string> &);

    REGISTER_COMMAND(reloadscripts, "Reloads server scripts!", cmd_reloadscripts,
                     info.usages = {"/reloadscripts"};
                     info.permissions = {"primebds.command.reloadscripts"};
                     info.aliases = {"rscripts", "rs"};);

    /// Reloads server scripts!
    static bool cmd_reloadscripts(PrimeBDS &plugin, endstone::CommandSender &sender,
                                  const std::vector<std::string> &args) {
        plugin.getServer().reloadData();
        sender.sendMessage("\u00a7aServer scripts reloaded");
        return true;
    }

} // namespace primebds::commands
